### PR TITLE
snapd-lib: fixes for UC 24 testing

### PR DESCRIPTION
**Pull request by Meulengracht**
Couple of fixes discovered while testing for modem-manager

### CODE REVIEW

File: snapd-lib/prepare-restore.sh [hunk 1]
```diff
@@ -610,10 +610,18 @@ prepare_project_each() {
     fixup_dev_random
 }
 
+is_testing_core() {
```
Comment (Meulengracht) [inline snapd-lib/prepare-restore.sh]: Logic ported from snapd

File: snapd-lib/state.sh [hunk 2]
```diff
@@ -8,6 +8,36 @@ SNAPD_ACTIVE_UNITS="$RUNTIME_STATE_PATH/snapd-active-units"
 # shellcheck source=tests/lib/systemd.sh
 . "$TESTSLIB/systemd.sh"
 
+# Set of helpers for checking if the test system is expected to be
```
Comment (Meulengracht) [inline snapd-lib/state.sh:11]: logic taken from snapd repo

File: snapd-lib/assertions/ubuntu-core-24-amd64.model [hunk 3]
```diff
@@ -1,48 +1,49 @@
 type: model
```
Comment (Meulengracht) [inline snapd-lib/assertions/ubuntu-core-24-amd64.model:1]: Had copied the stable model instead of dangerous - this is now tested better and works

**Review by alfonsosanchezbeato — APPROVED** [review]
LGTM, thanks!

**Pull request merged**